frontend/src/lib/dashboard/getUpdatedLayout.ts

```diff
@@ -0,0 +1,23 @@
+import { PANEL_TYPES } from 'constants/queryBuilder';
+import { Layout } from 'react-grid-layout';
+
+export const getUpdatedLayout = (layout?: Layout[]): Layout[] => {
+	let widgetLayout = layout;
+
+	// filter empty from i from i due to previous version of signoz
+	widgetLayout = layout?.filter((i) => i.i !== 'empty');
+
+	const seen = new Set();
+
+	// filter duplicate i values
+	widgetLayout = widgetLayout?.filter((i) => {
+		const duplicate = seen.has(i.i);
+		seen.add(i.i);
+		return !duplicate;
+	});
+
+	// filter EMPTY_WIDGET from i due to previous version of signoz
+	widgetLayout = widgetLayout?.filter((i) => i.i !== PANEL_TYPES.EMPTY_WIDGET);
+
+	return widgetLayout || [];
```

Comment:
I think this kind of code is wiping out the layout array. Assume the dashboard GET request failed for some reason. What would be the layout array on the client side? And if the backend works again after some time will there be PUT with the empty layout array?

Reply:
discussed offline need to find the case where layout is getting empty which is not related to this PR

Reply:
cc: @srikanthccv @YounixM resolving this thread